service/pokemon.go

```diff
@@ -22,6 +26,115 @@ func (sp PokemonService) GetPokemonByID(pokemons []model.Pokemon, ID int) (model
 
 }
 
+// GetPokemonWorker: Get Pokemons based on ID (odd or even) using Workers pools. 
+func (sp PokemonService)GetPokemonWorker(pokemons []model.Pokemon, t string, items int, itemsPerWorker int) ([]model.Pokemon){
+	var wg sync.WaitGroup 
+	retrievePokemons:=  make([]model.Pokemon,0)
+
+	// Workers
+	workers :=  8 //(items/itemsPerWorker) + 1
+	// Number of jobs. 
+	numJobs:=len(pokemons)
+
+	// Define channels for communication 
+	jobs := make(chan model.Pokemon, numJobs)
+	results := make(chan model.Pokemon, numJobs)
+	done := make(chan bool, workers) // channel to end working pool. 
+
+	// Create and run Workers to handle jobs. 
+	log.Println(" ******* Started Worker pool with", workers, "Workers *********")
+	wg.Add(workers)
+	go func (){
+		for w:=1 ; w <= workers; w++ {
+			go createWorker(w, jobs, results, &wg, t, done, itemsPerWorker)
+		}
+
+	}()
+
+	// Send jobs to Workers
+	for i:= 0 ; i < numJobs; i++{
+			jobs <- pokemons[i]
+	}
+	close(jobs)
+
+	// Receive results from workers.
+	for j:=0; j< numJobs; j++{
+		// Check if we retrieve the target item number. 
+		if len(retrievePokemons) == items {
+			// Send signals to end to every worker. 
+			done <- true
+			break
+		}
+		// Obtain results from Worker pool. 
+		r:= <- results
+		if( r != model.Pokemon{}) {
+			// Append result to slice (return value of method).
+			retrievePokemons =  append(retrievePokemons, r)
+		}
+	}
+
+
+	// wait for all Workers to finish execution to log and return results. 
+	wg.Wait() 
+	close(done)
+	time.Sleep(time.Second)
+	log.Printf("Results #: %v\n", len(retrievePokemons))
+	return retrievePokemons
+}
+
+// createWorker: Handle Worker logic and values to result channel. 
+func createWorker(id int, jobs <-chan model.Pokemon, results chan<- model.Pokemon, wg *sync.WaitGroup, t string, done chan bool, items int) {
+	count := 0
+	// range between channel job.
+	for j := range jobs {
+		log.Println("worker", id, "started  job: ", j)
+		
+		select{
+			case <-done: // Signal Worker to stop. 
+				wg.Done() 
+				log.Println("****** SHUTDOWN WORKER #", id, "With", count, " Jobs Done")
+				return
+			default:
+						//Worker processing the Job
+		time.Sleep(time.Second)
+		switch t {
+			// in case type is equal to odd. 
+			case "odd":
+				if (j.ID % 2) == 1 {
+					count = count + 1
+					results <- j
+					continue
+				}else{
+					results <- model.Pokemon{}
+				}
```

Comment:
If having a continue, there's no need to add the else statement.
```suggestion
				if (j.ID % 2) == 1 {
					count = count + 1
					results <- j
					continue
				}
				results <- model.Pokemon{}
```